src/hash_map.hh

```diff
@@ -274,6 +275,9 @@ struct HashMap
 
     const Item& item(size_t index) const { return m_items[index]; }
 
+    template<typename KeyType, typename = EnableIfHashCompatible<KeyType>>
+    auto keys() const { return m_items | transform([](auto& i) { return i.key; }) | gather<Container<KeyType, domain>>(); }
+
```

Comment:
I'd rather avoid adding this to hashmap, also the use of Container there seems wrong, we might not want the same container type for storing the hashmap and gathering the keys.